api/python/tests/integration/test_packages.py

```diff
@@ -1410,3 +1424,68 @@ def test_resolve_hash(self):
 
         with pytest.raises(QuiltException, match='Found multiple matches'):
             Package.resolve_hash(pkg_name, LOCAL_REGISTRY, hash_prefix)
+
+
+# The following tests were moved out of the PackageTest class to enable parametrization.
+# see (https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses)
+@pytest.mark.parametrize(
+    'update_policy, expected_one_byte',
+    [
```

Comment:
this is much better! thanks for cleaning this up. a few small things then we can merge:
1. we need to test nested directories (e.g. check what happens in nested/sub)
2. add a test where we don't provide an update_policy and show that it behaves just like `'incoming'`